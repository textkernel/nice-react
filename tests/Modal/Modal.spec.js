import React from 'react';
import { Modal } from '../../dist';

describe('<Modal />', () => {
    it('should render a simple Modal', () => {
        const wrapper = shallow(<Modal context="info">
            <Modal.Header>
                Modal header
            </Modal.Header>
            <Modal.Sub>
                Modal sub header
            </Modal.Sub>
            <Modal.Content>
                Modal content
            </Modal.Content>
            <Modal.Footer>
                Modal footer
            </Modal.Footer>
        </Modal>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render a Modal with custom classNames/styles', () => {
        const props = { 
            className: 'my-custom-classname',
            style: { top: 0 }
        };

        const wrapper = render(
            <Modal context="info" {...props}>
                <Modal.Header {...props}>
                    Modal header
                </Modal.Header>
                <Modal.Sub {...props}>
                    Modal sub header
                </Modal.Sub>
                <Modal.Content {...props}>
                    Modal content
                </Modal.Content>
                <Modal.Footer {...props}>
                    Modal footer
                </Modal.Footer>
            </Modal>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render a hidden, viewport centered Modal with overlay', () => {
        const mockCallback = jest.fn();
        const wrapper = shallow(<Modal context="info" viewportCenter overlay tall wide hidden>
            <Modal.Header onClose={ mockCallback }>
                Modal header
            </Modal.Header>
            <Modal.Sub>
                Modal sub header
            </Modal.Sub>
            <Modal.Content>
                Modal content
            </Modal.Content>
            <Modal.Footer>
                Modal footer
            </Modal.Footer>
        </Modal>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render a styled Modal', () => {
        const mockCallback = jest.fn();
        const wrapper = shallow(<Modal context="info" style={{ width: '600px', height: '600px' }}>
            <Modal.Header onClose={ mockCallback }>
                Modal header
            </Modal.Header>
            <Modal.Sub>
                Modal sub header
            </Modal.Sub>
            <Modal.Content>
                Modal content
            </Modal.Content>
            <Modal.Footer>
                Modal footer
            </Modal.Footer>
        </Modal>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render Modal.Header', () => {
        const mockCallback = jest.fn();
        const wrapper = shallow(<Modal.Header onClose={ mockCallback }>
            Modal header
        </Modal.Header>);
        expect(wrapper).toMatchSnapshot();

        wrapper.find('a.nice-modal-close').simulate('click', { preventDefault() {} });
        expect(mockCallback).toHaveBeenCalled();
    });

    it('should render Modal.Content', () => {
        const wrapper = shallow(<Modal.Content>
            Modal content
        </Modal.Content>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render Modal.Footer', () => {
        const wrapper = shallow(<Modal.Footer>
            Modal footer
        </Modal.Footer>);
        expect(wrapper).toMatchSnapshot();
    });
});
