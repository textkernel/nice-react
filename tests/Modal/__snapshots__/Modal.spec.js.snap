// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should render Modal.Content 1`] = `
<div
  className="nice-modal-body"
>
  Modal content
</div>
`;

exports[`<Modal /> should render Modal.Footer 1`] = `
<div
  className="nice-modal-footer"
>
  Modal footer
</div>
`;

exports[`<Modal /> should render Modal.Header 1`] = `
<div
  className="nice-modal-header"
>
  <a
    className="nice-modal-close"
    href="#"
    onClick={[Function]}
  >
    <i
      className="icon fa-times"
    />
  </a>
  <h4>
    Modal header
  </h4>
</div>
`;

exports[`<Modal /> should render a hidden, viewport centered Modal with overlay 1`] = `
<div
  className="nice-modal-overlay"
>
  <div
    className="nice-modal info viewport-center hidden tall wide"
    style={Object {}}
  >
    <div
      className="nice-modal-content"
    >
      <ModalHeader
        onClose={[Function]}
      >
        Modal header
      </ModalHeader>
      <ModalSub>
        Modal sub header
      </ModalSub>
      <Component>
        Modal content
      </Component>
      <ModalFooter>
        Modal footer
      </ModalFooter>
    </div>
  </div>
</div>
`;

exports[`<Modal /> should render a simple Modal 1`] = `
<div
  className="nice-modal-overlay"
>
  <div
    className="nice-modal info"
    style={Object {}}
  >
    <div
      className="nice-modal-content"
    >
      <ModalHeader
        onClose={null}
      >
        Modal header
      </ModalHeader>
      <ModalSub>
        Modal sub header
      </ModalSub>
      <Component>
        Modal content
      </Component>
      <ModalFooter>
        Modal footer
      </ModalFooter>
    </div>
  </div>
</div>
`;

exports[`<Modal /> should render a styled Modal 1`] = `
<div
  className="nice-modal-overlay"
>
  <div
    className="nice-modal info"
    style={
      Object {
        "height": "600px",
        "width": "600px",
      }
    }
  >
    <div
      className="nice-modal-content"
    >
      <ModalHeader
        onClose={[Function]}
      >
        Modal header
      </ModalHeader>
      <ModalSub>
        Modal sub header
      </ModalSub>
      <Component>
        Modal content
      </Component>
      <ModalFooter>
        Modal footer
      </ModalFooter>
    </div>
  </div>
</div>
`;
